### PR TITLE
Add logging defaults support from Metaflow profile.

### LINKED ISSUE #471: Expose logConfiguration options for jobs running on AWS Batch
By default, AWS Batch uses the `awslogs` log driver to ship logs for a job container to AWS CloudWatch.  However, this behavior [is configurable](https://docs.aws.amazon.com/batch/latest/APIReference/API_LogConfiguration.html), and even the `awslogs` driver itself [has configurability](https://docs.aws.amazon.com/batch/latest/userguide/using_awslogs.html) that may be of interest to Metaflow users.  

Example motivating usecases:
- specifying the log group or stream prefix may be useful for organizations that have schemes dictating how logs are collected in CloudWatch
- users of the Python standard logging library may benefit from the ability to configure multi-line pattern matching to ensure tracebacks and other multi-line log messages are interpreted as single events in CloudWatch.

The ask here is to expose `awslogs` logging config options, preferably in two ways:
1. as properties settable per step when using the `@batch` decorator
2. and, as is the case for other Batch configuration, as defaults in the Metaflow config profile.

Because Metaflow currently has an implicit dependency on logs making their way to CloudWatch, allowing users to specify log drivers other than `awslogs` may not be feasible today.  Still, this is something worth thinking about for the future.

### PR DESCRIPTION
**Pull request by Faakhir30**
## Summary

Added support for configuring default AWS Batch CloudWatch logging options using Metaflow profiles via a new `METAFLOW_LOG_OPTIONS` configuration variable.

## Motivation

While `@batch(log_options=...)` already allows per-step configuration, there was no way to set organization-wide defaults through Metaflow profiles or environment variables. This PR fills that gap.

Fixes #471 (the part about configuring defaults via profile).

## What’s changed

* Added `METAFLOW_LOG_OPTIONS` to `metaflow_config.py` to store default CloudWatch log options.
* Updated `configure_aws_batch()` to prompt for and validate these options during setup.
* Accepts comma-separated `key:value` pairs, for example:
  `awslogs-group:aws/batch/jobs,awslogs-stream-prefix:metaflow`
* Validates that each option contains a colon.
* Supports clearing the value by entering `none` or an empty string.
* Updated `BatchDecorator` to use `METAFLOW_LOG_OPTIONS` as the default for `log_options`.

## Testing

Added `test_configure_log_options.py` with coverage for:

* Accepting a single `key:value` pair
* Accepting multiple comma-separated pairs
* Rejecting values without a colon
* Rejecting incorrectly formatted lists
* Test reseting of `METAFLOW_LOG_OPTIONS` by passing explicit `none`.

## Design notes

* Validation only checks for `key:value` format instead of enforcing specific AWS keys, to remain flexible if AWS adds new options.
* Configuration is handled through `metaflow configure aws`, consistent with other Batch-related settings.
* If `METAFLOW_LOG_OPTIONS` is unset, behavior is unchanged.

### CODE REVIEW

**Comment by npow** [comment]
@greptile

File: metaflow/cmd/configure_cmd.py [hunk 1]
```diff
@@ -537,6 +538,38 @@ def configure_aws_batch(existing_env):
             default=existing_env.get("METAFLOW_SFN_DYNAMO_DB_TABLE"),
             show_default=True,
         )
+
+    # Configure logging of AWS Batch job outputs to CloudWatch.
+    while True:
+        default_val = ",".join(existing_env.get("METAFLOW_LOG_OPTIONS") or [])
```
[the remark this answers is not in the document]
Comment (Faakhir30) [reply metaflow/cmd/configure_cmd.py]: `existing_env["METAFLOW_LOG_OPTIONS"]` will never be a string, because I am always saving it as a `list`.

It can only be `None` unset case or `List[str]`.

cc @npow

**Comment by Faakhir30** [comment]
@npow Ready for review.

Regarding concern about unset behavior of `METAFLOW_BATCH_LOG_OPTIONS`. It works fine, I have tested locally with `metaflow configure aws`. Also added a test for passing `none` to reset.
